### ISSUE TITLE
The finder/importlib fails importing os.path

### ISSUE THREAD

**Issue by Traktormaster** [marcelotduarte/cx_Freeze#2646]
Building an application fails if it imports `os.path`.

**To Reproduce**
```python
import os.path

def main():
    print(os.path.join("asd", "xyz"))

if __name__ == "__main__":
    main()
```

Calling `cxfreeze pathy.py` fails with:
```
$ cxfreeze pathy.py
running build_exe
Traceback (most recent call last):
  File "/home/user/PycharmProjects/.venv/dev3/bin/cxfreeze", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/cli.py", line 246, in main
    setup(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/__init__.py", line 66, in setup
    return setuptools.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/setuptools/__init__.py", line 117, in setup
    return distutils.core.setup(**attrs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 183, in setup
    return run_commands(dist)
           ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/setuptools/_distutils/core.py", line 199, in run_commands
    dist.run_commands()
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 954, in run_commands
    self.run_command(cmd)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/setuptools/dist.py", line 991, in run_command
    super().run_command(command)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/setuptools/_distutils/dist.py", line 973, in run_command
    cmd_obj.run()
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/command/build_exe.py", line 278, in run
    freezer: Freezer = Freezer(
                       ^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/freezer.py", line 1258, in __init__
    Freezer.__init__(self, *args, **kwargs)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/freezer.py", line 135, in __init__
    self.finder: ModuleFinder = self._get_module_finder()
                                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/freezer.py", line 413, in _get_module_finder
    finder.add_base_modules()
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 719, in add_base_modules
    self.include_package("collections")
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 801, in include_package
    module = self._import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 700, in _scan_code
    self._scan_code(
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 268, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 424, in _load_module
    self._load_module_code(module, loader, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 474, in _load_module_code
    self._scan_code(module, deferred_imports)
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 662, in _scan_code
    imported_module = self._import_module(
                      ^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 302, in _import_module
    module = self._internal_import_module(name, deferred_imports)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 338, in _internal_import_module
    parent_module = self._internal_import_module(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 355, in _internal_import_module
    module = self._load_module(
             ^^^^^^^^^^^^^^^^^^
  File "/home/user/PycharmProjects/.venv/dev3/lib/python3.12/site-packages/cx_Freeze/finder.py", line 380, in _load_module
    spec = importlib.machinery.PathFinder.find_spec(name, path)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen importlib._bootstrap_external>", line 1537, in find_spec
  File "<frozen importlib._bootstrap_external>", line 1335, in __init__
  File "<frozen importlib._bootstrap_external>", line 1351, in _get_parent_path
AttributeError: module 'posixpath' has no attribute '__path__'. Did you mean: '__all__'?

```
**Expected behavior**
The build should succeed.

**Desktop (please complete the following information):**
 - Platform information: Arch Linux 6.6.54-1-lts x86_64 GNU/Linux
 - cx_Freeze version: 7.2.3
 - Python version: 3.12

**Additional context**
I have checked, the same issue is present on Windows 11 as well, the module name is "ntpath" instead of "posixpath".

**Comment by marcelotduarte** [comment]
I couldn't reproduce the error. I used your example, with the same names.
I tested with py310 (ubuntu), py312 (centos) and py312 on Windows.
I used a clean venv for each test.
For the py310 test, I used cx_Freeze 7.2.0 to 7.2.3 and also the 7.3.0-dev version.

**Comment by Traktormaster** [comment]
I've narrowed the issue down. There is a dependency installed from PyPI in the virtualenv that I'm trying to freeze that causes the issue. I did not even check, because it seemed to be rational to assume that no dependency would be pulled in by the sample script.

The [`path`](https://pypi.org/project/path/) package causes the problem.

Interestingly the freezing also fails with the same error if there are no imports at all:
```python
def main():
    print("hi!")


if __name__ == "__main__":
    main()
```

I looked into it, and I do not know what the `path` library does that is actually problematic. I'm also not sure why it is imported by the freezing process at all in the original sample, or the new one without import statements.
